itests/org.openhab.binding.mqtt.homie.tests/src/main/java/org/openhab/binding/mqtt/HomieImplementationTest.java

```diff
@@ -65,25 +65,26 @@
  *
  * @author David Graeff - Initial contribution
  */
+@NonNullByDefault
 public class HomieImplementationTest extends JavaOSGiTest {
     private static final String BASE_TOPIC = "homie";
     private static final String DEVICE_ID = ThingChannelConstants.testHomieThing.getId();
     private static final String DEVICE_TOPIC = BASE_TOPIC + "/" + DEVICE_ID;
 
-    private MqttService mqttService;
-    private MqttBrokerConnection embeddedConnection;
-    private MqttBrokerConnection connection;
+    private @NonNullByDefault({}) MqttService mqttService;
+    private @NonNullByDefault({}) MqttBrokerConnection embeddedConnection;
+    private @NonNullByDefault({}) MqttBrokerConnection connection;
     private int registeredTopics = 100;
 
-    private AutoCloseable mocksCloseable;
+    private @NonNullByDefault({}) AutoCloseable mocksCloseable;
 
     // The handler is not tested here, so just mock the callback
-    private @Mock DeviceCallback callback;
+    private @Mock @NonNullByDefault({}) DeviceCallback callback;
 
     // A handler mock is required to verify that channel value changes have been received
-    private @Mock HomieThingHandler handler;
+    private @Mock @NonNullByDefault({}) HomieThingHandler handler;
 
-    private ScheduledExecutorService scheduler;
+    private ScheduledExecutorService scheduler = new ScheduledThreadPoolExecutor(6);
```

Comment:
This scheduler instance isn't shutdown since it is immediately replaced by another instance in the `@BeforeEach` method.